### ISSUE TITLE
Should a edit/save/cancel component be provided by this addon

### ISSUE THREAD

**Issue by cah-briangantzler** [onechiporenko/ember-models-table#281]
* **I'm submitting a ...**

  - [ ] bug report
  - [x] feature request
  - [ ] support request

* **What is the current behavior?**
User has to implement the component to manage inline editing state

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

* **[Bug] What is the expected behavior?**

* **[Feature] What is the motivation / use case for changing the behavior?**
A component to maintain edit row state seems like something that would be common. However look and feel as well as actions may not be

* **[Support] What exactly you try to do?**

* **Please tell us about your environment:**

- `ember-models-table` version: 
- `ember` version: 
- `ember-data` version: 
- `ember-cli` version:  
- `node` version:
- `npm` or `yarn` version:

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

In the examples there is a component that is used to maintain the edit row state. It also handles save and cancel. 

If users what similar functionality, should they implement it themsevles or should this addon provide some default component to build on.

I think that if there was a component, it should be specified through column array as the example and not provide a boolean switch like allowInlineEdit. You would have to allow is it the first column, the last what is the header etc. The easiest solution is the user adds a column to the column array where they want it. The they can control most of what is needed. Then its in line with the example.

Basically, do you see the need to provide a component for use. If so what would you think this component may provide.

If we can come up with a consensus, I would be happy to create it.

**Comment by onechiporenko** [comment]
Hi, @cah-briangantzler. Definitely `models-table` must provide a default component for `edit` functionality.
How I see it:
* Component has three `actions`: `edit`, `save`, `cancel` for three buttons.
* This actions calls "external" closure actions provided to the component (may be with same names - `edit`, `save`, `cancel`).
* This actions checks if result from "external" actions is Promise or not and wait for truly result.
* Component shows three buttons with labels like "Edit", "Save", "Cancel". All of them are provided from `messages` (in the `models-table`).
* Classes from `themeInstance` for buttons are added for them.

_Some of this items are already implemented in your `edit-row`._

Valid yuidoc-comments with types, defaults, usage examples etc are welcome :)

**Comment by cah-briangantzler** [comment]
Ok, should just be a couple of changes to copy the one in the example over to the main addon.

Any suggestion for name? or is `edit-row` ok?

**Comment by onechiporenko** [comment]
`models-table/cell-edit-toggle` or something like this would be good.
